### PR TITLE
Enable references tests and adjust them for zero sized unit type

### PR DESCRIPTION
**Pull request by ironcev**
## Description

This PR enables `language/references` and `stdlib` tests that were disabled because of the #6898. It also adjusts the `language/references` tests to the change introduced in #7509, which changes unit to be a zero-sized type.

## Checklist

- [x] I have linked to any relevant issues.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Comment by ironcev** [comment]
:+1:

File: test/src/e2e_vm_tests/test_programs/should_pass/language/references/passing_and_returning_references_to_and_from_functions/test.toml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# TODO Enable this test once https://github.com/FuelLabs/sway/issues/6898 is fixed.
 category = "disabled"
```
[the remark this answers is not in the document]
Comment (ironcev) [reply test/src/e2e_vm_tests/test_programs/should_pass/language/references/passing_and_returning_references_to_and_from_functions/test.toml]: Good catch! Fixed!

File: test/src/e2e_vm_tests/test_programs/should_pass/language/references/referencing_local_vars_and_values/src/main.sw [hunk 2]
```diff
@@ -229,10 +228,10 @@ fn test_not_inlined() {
     reference_local_var_and_value_not_inlined::<raw_ptr>();
     reference_local_var_and_value_not_inlined::<raw_slice>();
     
+    reference_zero_sized_local_var_and_value_not_inlined::<()>();
     reference_zero_sized_local_var_and_value_not_inlined::<EmptyStruct>();
     reference_zero_sized_local_var_and_value_not_inlined::<[u64;0]>();
```
[the remark this answers is not in the document]
Comment (ironcev) [reply test/src/e2e_vm_tests/test_programs/should_pass/language/references/referencing_local_vars_and_values/src/main.sw:233]: Good catch! Fixed!

**Review by xunilrj — APPROVED** [review]
